jni/src/common/Foreign/JNI/Unsafe.hs

```diff
@@ -376,6 +380,7 @@ attachCurrentThreadAsDaemon = do
       [CU.exp| jint {
         (*$(JavaVM* jvm))->AttachCurrentThreadAsDaemon($(JavaVM* jvm), (void**)&jniEnv, NULL)
       } |]
+    readIORef mainThreadClassLoader >>= setCurrentThreadClassLoader
```

Comment:
@m, I'm struggling with a design choice here. With the change to delete global references in a background thread we need to use the threaded runtime, and tests in jvm-streaming will fail in the threaded runtime because we need to set the context ClassLoader on some of the OS threads that we use.

I proposed to @nbacquey to set it here, but I'm noticing that this runs against the philosophy of keeping the bindings in jni simple.

Maybe the best place to put them would be in a new convenience package inline-java-threads, where we can provide a version of `runInAttachedThread` and `withJVM` that deal with setting context class loaders for the user. Otherwise, the user would need to be responsible for doing this in each application.

Any thoughts?